### PR TITLE
[Improvement](hive view) support inner name like _c0,_c1 for derived column in inline view for hive view

### PR DESCRIPTION
**Pull request by Yulei-Yang**
update planer,  support queries on hive view which use inner name, act like hive. sql e.g.:
```sql
#hive
create view v1 as
select r_name,
       `_c1` as col1
from
  (select r_name,
          count(*)
   from region
   group by r_name) t1;

#doris
select * from hms_catalog.hms_db.v1;

```
before this pr, doris throw error "Unknown column '_c1' in 'table list'" when execute this sql.

**Ilustrate**
Num `N`  in inner name `_cN` starts at 0, and is the index in all columns of a inline view, not only derived/calculated columns.

**Restriction**
Inner name can only be used for derived/calculated column, same with hive. In the first sql example above, column type in inner query is not  derived/calculated column, below sql is error:
```sql
mysql> select _c0, _c1 from (SELECT `type`, count(*) FROM hms_catalog.hms_db.`table1` GROUP BY `type`) t1;            
ERROR 1054 (42S22): errCode = 2, detailMessage = Unknown column '_c0' in 'table list'
```

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
run buildall

**Review by lide-reed — DISMISSED** [review]
LGTM

**Review by morrySnow — CHANGES_REQUESTED** [review]
according to offline discussion, we should
1. Restrict this change to take effect only for hive views
2. change both Nereids and legacy planner

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
> 2\. Nereids

1. done
2. will support in nereids planner at next pr

**Comment by Yulei-Yang** [comment]
rerun p0

**Review by morrySnow — COMMENTED** [review]
i think this PR only process legacy planner? please add test case to ensure it could work as u expected

**Comment by morrySnow** [comment]
BTW, i don't think hive view could work well in Doris, since their syntax is not same

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
> according to offline discussion, we should
> 
> 1. Restrict this change to take effect only for hive views
> 2. change both Nereids and legacy planner

done

**Comment by Yulei-Yang** [comment]
> i think this PR only process legacy planner? please add test case to ensure it could work as u expected

done

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run feut

**Comment by Yulei-Yang** [comment]
run feut

**Comment by Yulei-Yang** [comment]
run buildall

**Comment by Yulei-Yang** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundAlias.java [hunk 1]
```diff
@@ -89,4 +89,8 @@ public UnboundAlias withChildren(List<Expression> children) {
     public Optional<String> getAlias() {
         return alias;
     }
+
+    public void setAlias(Optional<String> alias) {
+        this.alias = alias;
+    }
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/analyzer/UnboundAlias.java:95]: this is a mistake of `UnboundAlias`. In Nereids all Expression and plan should be Immutable. So we should let alias be final. and add a function `withAlias` like `withChildren` to generate a new `UnboundAlias` object with new alias

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java [hunk 2]
```diff
@@ -328,13 +333,57 @@ private Plan parseAndAnalyzeHiveView(
...
+    }
+
     private Plan parseAndAnalyzeView(TableIf view, String ddlSql, CascadesContext parentContext) {
         parentContext.getStatementContext().addViewDdlSql(ddlSql);
         Optional<SqlCacheContext> sqlCacheContext = parentContext.getStatementContext().getSqlCacheContext();
         if (sqlCacheContext.isPresent()) {
             sqlCacheContext.get().addUsedView(view, ddlSql);
         }
         LogicalPlan parsedViewPlan = new NereidsParser().parseSingle(ddlSql);
+        addHiveViewDerivedColumn(parsedViewPlan);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:386]: this function use for both doris view and hive view. so also process doris' view?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java [hunk 3]
```diff
@@ -328,13 +333,57 @@ private Plan parseAndAnalyzeHiveView(
...
+            for (NamedExpression project : projects) {
+                counter++;
+                // only set inner name for derived column
+                if (project instanceof UnboundAlias) {
+                    UnboundAlias aliasProject = (UnboundAlias) project;
+                    if (aliasProject.getAlias().isPresent()) {
+                        continue;
+                    }
+                    if (aliasProject.child() instanceof UnboundFunction) {
+                        aliasProject.setAlias(Optional.of("_c" + counter));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:352]: if we repalce all alias with '_cx', what will happen for this sql
```sql
SELECT a1, a2 FROM (SELECT id AS a1, count(*) AS a2 FROM t1 GROUP BY id) inline_view
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java (hunk 2 above)
Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/BindRelation.java:386]: fixed

**Pull request closed without merging**